### PR TITLE
Add meta fields section to ReadTheDocs

### PR DESCRIPTION
**Pull request by pmantica1**
In this PR, I add meta fields to readthedocs.

### CODE REVIEW

File: docs/source/language_specification/schema_types.rst [hunk 1]
```diff
@@ -298,3 +308,210 @@ and discarded.
 In this query, the :code:`out_Entity_Related` is of :code:`Entity` type.
 However, the query only wants to return results where the related entity
 is a :code:`Species`, which :code:`... on Species` ensures is the case.
+
+
+Meta fields
+-----------
+
+Meta fields are fields that do not represent a property/column in the underlying vertex type.
+They are also an advanced compiler feature. To use them, one needs to understand how certain
```
Comment (obi1kenobi) [inline docs/source/language_specification/schema_types.rst]: You've described what they aren't, but what are they? :)

Also, "certain compiler directives" is pretty vague, any reason for that? Why not just say "Before continuing, readers should familiarize themselves with the various query directives supported by the compiler." and throw in a link?

Comment (pmantica1) [reply docs/source/language_specification/schema_types.rst]: Well I was thinking that you need to understand @fold to understand how to use the _x_count meta field, (technically also @output), but the way you phrase it makes sense.

I cannot think of a better way to describe meta fields. I looked at the GraphQL spec and the GraphQL website for inspiration and couldn't find a definition of what a "meta fied" in GraphQL is.

Comment (obi1kenobi) [reply docs/source/language_specification/schema_types.rst]: Please be mindful that @-syntax is how GitHub allows you to tag users. So unless you surround your directives with backticks (e.g. `@fold`), you are tagging random GitHub users and probably causing notifications on their end.

For the best description, maybe we can sit down with @JulianGoetz and figure out how to describe this idea precisely yet succinctly.

Comment (pmantica1) [reply docs/source/language_specification/schema_types.rst]: Do you mind approving this PR though beforehand? This PR is for sure an improvement over existing documentation. I can open an issue for the lack of a precise definition of meta fields.

**Comment by pmantica1** [comment]
Here is the issue: https://github.com/kensho-technologies/graphql-compiler/issues/604

**Pull request merged**